core/src/main/java/org/apache/druid/coordination/CommonCallback.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.coordination;
+
+public interface CommonCallback
```

Comment:
why not just use `Runnable`? There is even a `Runnables.getNoopRunnable()`

Reply:
I think a meaningful name is better for code readiness through the method arguments

Reply:
I don't feel super strongly about this, while I find it pretty obvious what `Runnable` does and don't think a new interface adds much here, I can see your point as well. 

I would agree that if we insist on using a separate interface that it should have a better name more indicative of its function.

Reply:
Change the name to `SegmentLazyLoadFailCallback` as you guys suggested. Thanks for review :)